source/common/ratelimit/ratelimit_impl.cc

```diff
@@ -100,5 +101,35 @@ ClientPtr GrpcFactoryImpl::create(const Optional<std::chrono::milliseconds>& tim
       timeout)};
 }
 
+class RateLimitSpanFinalizer : public Tracing::SpanFinalizer {
+public:
+  RateLimitSpanFinalizer(const pb::lyft::ratelimit::RateLimitRequest& request,
+                         const pb::lyft::ratelimit::RateLimitResponse* response)
+      : request_(request), response_(response) {}
+
+  void finalize(Tracing::Span& span) {
+    if (response_ != nullptr) {
```

Comment:
any tests covering this?
can you run coverage to make sure we have 100 for the newly added code.